executor/aggregate.go

```diff
@@ -922,6 +974,17 @@ func (e *HashAggExec) execute(ctx context.Context) (err error) {
 		for j := 0; j < e.childResult.NumRows(); j++ {
 			groupKey := string(e.groupKeyBuffer[j]) // do memory copy here, because e.groupKeyBuffer may be reused.
 			if !e.groupSet.Exist(groupKey) {
+				if atomic.LoadUint32(&e.spillMode) == 1 && e.groupSet.Count() > 0 {
+					e.spillChunk.Append(e.childResult, j, j+1)
```

Comment:
Can we use `Chunk.sel` to optimize this if-block?
1. We can check `e.groupSet.Exist(groupKey)` and build the `sel` firstly, and then invoke `e.spillChunk.Append` based on the `sel`.
2. Further, if `len(sel) == len(e.childResult)`, we can invoke `e.listInDisk.Add(e.childResult)` directly.

Reply:
Addressed.